### PR TITLE
Use local ref count table to speed up deletes

### PR DESCRIPTION
**Pull request by dergraf**
Bashobench using separate delete and put operations (50/50)
 Bashobench using separate delete and put operations (50/50)

Bashobench using  put and delete within same operations
Bashobench using  put and delete within same operations

### CODE REVIEW

**Comment by dergraf** [comment]
@larshesel please review. It's valid to merge, NO bcbreak!

File: src/vmq_lvldb_store.erl [hunk 1]
```diff
@@ -421,14 +417,34 @@ check_delete_message(Bucket, {ok, Key}, Itr, WriteOpts, {PivotMsgRef, PivotMP, H
...
+    try ets:update_counter(Refs, MsgRef, -1) of
+        0 ->
+            ets:delete(Refs, MsgRef),
+            0;
+        V ->
+            V
+    catch
+        _:_ ->
+            not_found
+    end.
```
Comment (larshesel) [inline src/vmq_lvldb_store.erl:450]: is it necessary with the try catch? What could trigger an exception here?

**Comment by larshesel** [comment]
LGTM (although I'm curious about the try catch around the ets:update_counter call)

File: src/vmq_lvldb_store.erl (hunk 1 above)
Comment (dergraf) [inline src/vmq_lvldb_store.erl:450]: Update_counter gives you a badarg if the counter doesn't exist. I think it makes sense to catch this exception and don't crash the msgstore as a restart typically won't help in such error conditions.

**Pull request merged**